src/app/calculator/furnaces/atmosphere/atmosphere.service.ts

```diff
@@ -0,0 +1,276 @@
+    let currentBaselineCopy = JSON.parse(JSON.stringify(currentBaselineData));
+    this.modificationData.next(currentBaselineCopy);
+  }
+
+  updateDataArray(data: AtmosphereLoss, index: number, isBaseline: boolean) {
+    let dataArray: Array<AtmosphereLoss>;
+    if (isBaseline) {
+      dataArray = this.baselineData.getValue();
+    } else {
+      dataArray = this.modificationData.getValue();
+    }
+    // dataArray won't exist during reset cycle w/ multiple subjects emitting
+    if (dataArray && dataArray[index]) {
+      dataArray[index].name = data.name;
+      dataArray[index].hoursPerYear = data.hoursPerYear;
+      dataArray[index].fuelCost = data.fuelCost;
+      dataArray[index].availableHeat = data.availableHeat;
+      dataArray[index].energySourceType = data.energySourceType;
+      dataArray[index].atmosphereGas = data.atmosphereGas;
+      dataArray[index].specificHeat = data.specificHeat;
+      dataArray[index].inletTemperature = data.inletTemperature;
+      dataArray[index].outletTemperature = data.outletTemperature
+      dataArray[index].flowRate = data.flowRate;
+      dataArray[index].correctionFactor = data.correctionFactor;
+      dataArray[index].heatLoss = data.heatLoss;
+    }
+
+     if (isBaseline) {
+      this.baselineData.next(dataArray);
+    } else {
+      this.modificationData.next(dataArray);
+    }
+  }
+
+    
+  removeLoss(i: number) {
+    let currentBaselineData: Array<AtmosphereLoss> = this.baselineData.getValue();
+    currentBaselineData.splice(i, 1);
+    this.baselineData.next(currentBaselineData);
+    let currentModificationData: Array<AtmosphereLoss> = this.modificationData.getValue();
+    if (currentModificationData) {
+      currentModificationData.splice(i, 1);
+      this.modificationData.next(currentModificationData);
+    }
+  }
+  
+  addLoss(hoursPerYear: number, modificationExists: boolean) {
+    let currentBaselineData: Array<AtmosphereLoss> = JSON.parse(JSON.stringify(this.baselineData.getValue()));
+    let index = currentBaselineData.length;
+    let baselineObj: AtmosphereLoss = this.initDefaultLoss(index, hoursPerYear);
+    baselineObj.fuelCost = currentBaselineData[index - 1].fuelCost;
+    baselineObj.availableHeat = currentBaselineData[index - 1].availableHeat;
+    currentBaselineData.push(baselineObj)
+    this.baselineData.next(currentBaselineData);
+    
+    if (modificationExists) {
+      let currentModificationData: Array<AtmosphereLoss> = this.modificationData.getValue();
+      let modificationObj: AtmosphereLoss = this.initDefaultLoss(index, hoursPerYear);
+      
+      // Set loss operational constants
+      modificationObj.fuelCost = currentBaselineData[index - 1].fuelCost;
+      modificationObj.availableHeat = currentBaselineData[index - 1].availableHeat;
+
+      currentModificationData.push(modificationObj);
+      this.modificationData.next(currentModificationData);
+    }
+  }
+
+  generateExampleData(settings: Settings) {
```

Comment:
settings are unused currently. Should example data be converted depending on units of measure?